CDCgov/prime-reportstream
# 11113: add HD->Organization->HD mappings

This PR completes the HL7->FHIR->HL7 for HD/Location, HD/Organization, XON/Organization

Test Steps:
1. Review the FHIR inventory
  - https://docs.google.com/spreadsheets/d/1-cdMWepRbnqJOVVZ-y70Qv9Ub7evClmbX9gYO0kHmpo/edit#gid=0
  - https://docs.google.com/spreadsheets/d/1thMC8Et_hA-VC-DyerLrpNDcNdzPOIql9Z7SP7u9h_w/edit#gid=0
  - https://docs.google.com/spreadsheets/d/126xjG_YCcKYzO7f10dtlBOBUMNZZXkteoE2zNu-_J40/edit#gid=0
3. Review the test files and make sure all the conditions are covered

## Changes
- Completes the HD -> Organization -> HD mapping
- Completes the HD -> Location -> HD mapping
- Completes the XON -> Organization -> XON mapping

## Checklist

### Testing

### Process

## Linked Issues
- #11113

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #11113 (This PR only addresses some of the datatype mappings used by MSH. We will revisit the MSH story once all DataType mappings have been completed. See: https://github.com/CDCgov/prime-reportstream/issues/11655)

## Specific Security-related subjects a reviewer should pay specific attention to
